File: services/api/app/schemas/share_intake.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class ShareIntakeCreate(BaseModel):
    source_url: Optional[str] = None
    shared_text: Optional[str] = None
    title: Optional[str] = None
    notes: Optional[str] = None


class ShareIntakeResponse(ShareIntakeCreate):
    share_id: UUID
    source_domain: Optional[str] = None
    status: str
    detected_address: Optional[str] = None
    matched_property_id: Optional[str] = None
    created_property_id: Optional[str] = None
    created_at: datetime
    processed_at: Optional[datetime] = None
    error_message: Optional[str] = None

File: apps/web/src/types/shareIntake.ts
export type ShareIntakeStatus = "pending" | "processing" | "completed" | "needs_input" | "error";

export type ShareIntakeCreate = {
  source_url?: string;
  shared_text?: string;
  title?: string;
  notes?: string;
};

export type ShareIntake = ShareIntakeCreate & {
  share_id: string;
  source_domain?: string;
  status: ShareIntakeStatus;
  detected_address?: string;
  matched_property_id?: string;
  created_property_id?: string;
  created_at: string;
  processed_at?: string;
  error_message?: string;
};
